src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1252,9 +1252,12 @@ private static TypeSymbol GetParameterType(int argIndex, MemberAnalysisResult re
                                                        out okToDowngradeToNeither);
                 }
 
+                var type1Normalized = type1.NormalizeTaskTypes(Compilation);
+                var type2Normalized = type2.NormalizeTaskTypes(Compilation);
```

Comment:
That's funny! Even if we fix this particular problem (to consider two different task-like types as equivalent even in the absence of System.Threading.Tasks.Task/Task<T>), the compiler is still hard-coded to prefer Task in certain situations e.g.

```csharp
void f<T>(Func<T> lambda)
f(async () => 3);
```

I think it's a mistake that the compiler does this inference, but we're stuck with it. Imagine if we came up with a set of libraries which removed Task completely, and only used other tasks, as a way to force the compiler never to do these dangerous inferences!!!

Still it's a far-removed way-out scenario. Task has found its way into the deepest part of the BCL and will never be removed.